bms-monitor.js

```diff
@@ -1,18 +1,39 @@
-const {expect} = require('chai');
+const {limits}=require('./inputParameterLimits');
+const {outPutLogs}=require('./outputLogs');
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
+function batteryStatus(limits,outPutLogs)
+{
+        // function to know if breach is Higher or lower than given limits
+    function checkBreach(input,parameterRange,inputParameterType){
+            if(input<parameterRange.minimum)
+            {
+             return outPutLogs[inputParameterType].low;
+            }
+            else if(input>parameterRange.maximum){
+             return outPutLogs[inputParameterType].high;
+            }
+            return '';
+     }
+     
+    function batteryIsOk(temperature, soc, charge_rate) {
+
+        const temperatureErrorLog=checkBreach(temperature,limits.temperature,'temperature');
+        const socErrorLog=checkBreach(soc,limits.soc,'soc');
+        const chargeRateErrorLog=checkBreach(charge_rate,limits.chargeRate,'chargeRate');
+      
+        const errorLogs=[temperatureErrorLog,socErrorLog,chargeRateErrorLog]

```

Comment:
Nice way to combine multiple parameters into a single result :+1: 
Advantage is, the complexity will not increase when we add more parameters